apps/route-checker/src/app/services/pull.listener.service.ts

```diff
@@ -0,0 +1,29 @@
+import {EventData, EventType} from "../types/types";
+import interestingPlacesService from "./interesting.places.service";
+import mapService from "./map.service";
+
+class PullListenerService {
+
+    async handleEvents(type: string, event: EventData) {
+        if (EventType.CREATED !== type && EventType.UPDATED !== type)
+            return;
+        if (!event?._id?.length || !event?.coordinates?.length)
+            return;
+        if (EventType.UPDATED === type)
+            await interestingPlacesService.removeInterestingPlaces(event._id);
+        mapService.findPlacesByBBox(event.coordinates).then(response => {
+            if (!this.isOK(response))
+                return;
+            response.data.forEach(value => value.routeId = event._id);
+            interestingPlacesService.createInterestingPlaces(response.data);
+        });
```

Comment:
I would change this piece of code so it's easier to understand for the rest of the team. How does it look to you the following proposal?

Firstly, we should unify the parameters that each handler should receive, as there will be more handlers added. I would create one function per event called like: `routeCreatedHandler(), routeUpdatedHandler()...` and all of them should receive the `data` attribute from the pubSub message.

We could have an object like this to check which handler do we need in each case:

```typescript
const eventsHandlers = {
  [EventType.RouteCreated]: routeCreatedHandler,
  [EventType.RouteDeleted]: routeDeletedHandler,
}
```

and a simple function to get the event handler in each case:

```typescript
export function getEventHandler (attributes: MessageAttributes): (payload: string) => Promise<void> {
  const { type } = attributes

  return eventsHandlers[type]
}
```

and then just call the resultant function from the previous one with:

```
const eventHandler = getEventHandler(message.attributes)
await eventHandler(JSON.parse(message.data))
```